modules/tinymce/src/plugins/quickbars/test/ts/browser/KeyboardShortcutContextualMenuTest.ts

```diff
@@ -0,0 +1,33 @@
+import { FocusTools, Keys } from '@ephox/agar';
+import { describe, it } from '@ephox/bedrock-client';
+import { SugarDocument } from '@ephox/sugar';
+import { TinyHooks, TinySelections, TinyContentActions, TinyUiActions } from '@ephox/wrap-mcagar';
+
+import Editor from 'tinymce/core/api/Editor';
+import Plugin from 'tinymce/plugins/quickbars/Plugin';
+
+describe('browser.tinymce.plugins.quickbars.KeyboardShortcutContextualMenuTest', () => {
+  const hook = TinyHooks.bddSetup<Editor>({
+    plugins: 'quickbars',
+    quickbars_insert_toolbar: 'quicktable image media codesample',
+    quickbars_selection_toolbar: 'bold italic underline | formatselect | bullist numlist | blockquote quicklink',
+    contextmenu: 'undo redo | inserttable | cell row column deletetable | help',
+    base_url: '/project/tinymce/js/tinymce',
+  }, [ Plugin ]);
+
+  const doc = SugarDocument.getDocument();
+
+  it('TINY-2884: keypress should focus on contextual menu', async () => {
+    const editor = hook.editor();
+    editor.setContent('<p>aaaaaaa</p>');
+    editor.focus();
+    TinySelections.setSelection(editor, [ 0, 0 ], 0, [ 0, 0 ], 4);
+    // safari would need this or else won't pass
+    setTimeout(() => {
+      TinyContentActions.keystroke(editor, 120, { ctrl: true });
+    }, 40);
```

Comment:
As I said before, this is a very large red flag... there must be something else going on as we should almost never have a setTimeout like this in the tests. So either investigate this more please or pull one of us in to help diagnose what's going on.

Reply:
https://github.com/tinymce/tinymce/pull/7533/commits/9b15ad55debfdf08d9ca4e128a48fda400c8d171